examples/spark/images/master/start.sh

```diff
@@ -14,6 +14,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export SPARK_MASTER_PORT=${SPARK_MASTER_SERVICE_PORT:-7077}
+. /start-common.sh
+
+echo "$(hostname -i) spark-master" >> /etc/hosts
+export SPARK_LOCAL_HOSTNAME=spark-master
+export SPARK_MASTER_IP=spark-master
+
 /opt/spark/sbin/start-master.sh
 tail -F /opt/spark/logs/*
```

Comment:
+1 linking the pod lifecycle to the spark-master's, instead of to tail's

in fact, i thought there was a TODO comment about that in the original pr. the only downside i could think of for linking by executing the main class was that the spark upstream folks might change the wrapper to do more setup. revving the spark version will just require some care in inspecting start-master.sh changes.